models/project/issue.go

```diff
@@ -102,7 +102,37 @@ func MoveIssuesOnProjectBoard(ctx context.Context, board *Board, sortedIssueIDs
 	})
 }
 
-func (b *Board) removeIssues(ctx context.Context) error {
-	_, err := db.GetEngine(ctx).Exec("UPDATE `project_issue` SET project_board_id = 0 WHERE project_board_id = ? ", b.ID)
+func (b *Board) moveIssuesToAnotherColumn(ctx context.Context, columnID int64) error {
+	if b.ID == columnID {
+		return nil
+	}
+	_, err := db.GetEngine(ctx).Exec("UPDATE `project_issue` SET project_board_id = ? WHERE project_board_id = ? ", columnID, b.ID)
 	return err
 }
+
+// MoveColumnsOnProject moves or keeps issues in a column and sorts them inside that column
+func MoveColumnsOnProject(ctx context.Context, project *Project, sortedColumnIDs map[int64]int64) error {
+	return db.WithTx(ctx, func(ctx context.Context) error {
+		sess := db.GetEngine(ctx)
+
+		columnIDs := make([]int64, 0, len(sortedColumnIDs))
+		for _, columnID := range sortedColumnIDs {
+			columnIDs = append(columnIDs, columnID)
+		}
+		count, err := sess.Table(new(Board)).Where("project_id=?", project.ID).In("id", columnIDs).Count()
+		if err != nil {
+			return err
+		}
+		if int(count) != len(sortedColumnIDs) {
+			return fmt.Errorf("all issues have to be added to a project first")
```

Comment:
Error message is wrong when issues across multiple projects have been queried

Reply:
f280c9b